calico_node/startup/startup.go

```diff
@@ -267,8 +268,12 @@ func configureIPsAndSubnets(node *api.Node) {
 		adm := os.Getenv("IP_AUTODETECTION_METHOD")
 		cidr := autoDetectCIDR(adm, 4)
 		if cidr != nil {
-			// We autodetected an IPv4 address so update the value in the node.
-			node.Spec.BGP.IPv4Address = cidr
+			// We autodetected an IPv4 address, if we have no previous IP, or it is different than the previous IP,
+			// we need to update it and check for conflicts.
+			if node.Spec.BGP.IPv4Address == nil || !node.Spec.BGP.IPv4Address.IP.Equal(cidr.IP) {
```

Comment:
Hm, this only checks the IP part of the CIDR, not the netmask.

I think we might want to set `node.Spec.BGP.IPv4Address = cidr` always - it's possible that the IP stays the same but the subnet changes.